### DISCUSSION TITLE
About security

### DISCUSSION THREAD

**Discussion by n1majne3** [dani-garcia/vaultwarden#4202] [Q&A] [answered]
Hello
I have been running Vaultwarden on my NAS at home, it is great. but I have some questions about the security, I don't have my public IP, but I access it by Cloudflare tunnel and 3rd party FRP(for backup access ). I worry that even though those services helped me handle HTTPS automatically, I am not sure if they can inspect my traffic. If they do, will my password leak? Thanks.

**Comment by kpfleming** [comment]
If you use third parties to provide HTTPS connections to your server, those third parties are definitely able to inspect the traffic between the clients and your server.

To understand what this may or may not expose, take a few moments to read the [Bitwarden white paper](https://bitwarden.com/help/bitwarden-security-white-paper/) on security.

**Reply by tessus** [reply, marked as answer]
> If you use third parties to provide HTTPS connections to your server, those third parties are definitely able to inspect the traffic between the clients and your server.

This is true, but since the data is always en-/decrypted on the client, it doesn't really matter. Not that I'd support others being able to read any data that isn't required.

To answer the OP's question

> If they do, will my password leak?

No. They will only see encypted versions of the passwords. Same, as if you inspect the vw database.
